umbraco/The-Starter-Kit
# Update to work with Umbraco 11

## Details
- Created a new version of the Starter kit - v11;
- Updated dependencies:
  - Umbraco - 11.0.0-rc5;
  - OpenStreetMap - 5.1.0;
- Other:
  - Fixed the start-up project for VS in the .sln file;
  - Dotnet sdk version is updated to `7.x` for the pipeline;
  - Created a new **package.zip** containing the replacement of old/legacy prop editors (Media Picker (legacy), Grid layout and Nested content);
  - Added new views and css file needed for the block grid;
    - Everything in _Views/Partials/blockgrid/Components/_ comes from installing the Demo Block Grid; 
    - _umbraco-starterkit-blockgrid.css_ is used for formatting the grid when the content is rendered;
  - Updated some document-type templates to render the grid correctly;
  - Removed conn string and provider name in appsettings.json since we have to go through the installation anyway;
  - Removed unused files related to the old Grid layout - the whole **grid** folder in Partial Views;
  - Updated README;
  - Fixed non-working link (embedded video) in one of the Grid columns in "About Us" content node to https://youtu.be/3bSHnMZF9xI.

## Test
- Start up the starter kit from this branch and make sure you can navigate around;
- Open the command-line in the `Umbraco.SampleSite` folder and execute the following commands:
  - `dotnet restore`;
  - `dotnet build`;
  - `dotnet pack`;
- Find the _Umbraco.TheStarterKit.11.0.0-rc.nupkg_  in the `bin` folder;
- Use the latest dotnet umbraco template (11.0.0-rc5) and add the starter kit nuget package; 
- Make sure Grid layout or Nested content are not used in any doc type - you can check the Data types folder in the backoffice;
- Make sure the 3 blog posts appear on the _Home_ route and you can visit each;

![image](https://user-images.githubusercontent.com/21998037/202175057-66ace8a2-df0a-459c-8c46-361720a01cd6.png)
  
- Visit the `/about-us` and verify that the content is displayed in 2 columns (using the block grid)
  - Similar situation with `/blog/another-one/` - also changed to use the new Block grid;
-  `products/biker-jacket/` is using the Block List instead of Nested content for the Features group, but the content should look the same as before;
- Check if you can change the Hero Background and Logo images on the Home content node.
